editoast/src/views/timetable/stdcm.rs

```diff
@@ -464,17 +575,25 @@ fn get_earliest_step_tolerance_window(data: &STDCMRequestPayload) -> u64 {
         .unwrap_or(0)
 }
 
-/// Computes the simulation run time
-/// Returns an enum with either the result or a SimulationResponse if it failed
-async fn get_simulation_run_time(
+/// Returns a `Result` containing:
+/// * `SimulationTimeResult` - The result of the simulation time calculation.
+/// * `TrainSchedule` - The generated train schedule based on the provided data.
+/// * `SimulationResponse` - Simulation response.
+/// * `PathfindingResult` - Pathfinding result.
+async fn simulate_train_run(
     db_pool: Arc<DbConnectionPoolV2>,
     valkey_client: Arc<ValkeyClient>,
     core_client: Arc<CoreClient>,
     data: &STDCMRequestPayload,
     infra: &Infra,
     rolling_stock: &RollingStockModel,
     timetable_id: i64,
-) -> Result<SimulationTimeResult> {
+) -> Result<(
+    SimulationTimeResult,
+    TrainSchedule,
+    SimulationResponse,
+    PathfindingResult,
+)> {
```

Comment:
That's a big tuple, I would rather make a type out of it. Grouping the input (`TrainSchedule`) with the results (`SimulationResponse` and `PathFindingResult`) in a `struct` feels like it would make sense.

As for `SimulationTimeResult`, since it's a by-product of `SimulationResponse`, I would rather let the caller deal with it. What do you think? (cc @leovalais)

Reply:
See https://github.com/OpenRailAssociation/osrd/pull/9434